src/basic/schema.ts

```diff
@@ -58,8 +59,15 @@ export const publicKeyCredentialWithAttestation: Schema = {
   response: required({
     clientDataJSON: required(convert),
     attestationObject: required(convert),
+    transports: derived(
+      copy,
+      (response: any) => response.getTransports?.() || [],
```

Comment:
Review note: I was a little concerned that empty `transports` field might mess with the authentication prompt, but it doesn't seem to be an issue. I'm having trouble finding a place to back that up in the spec, though, so I'm just going off of Chrome/FirefoxSafari on macOS.

Reply:
The backing for this in the spec is in the [definition of the `getTransports()` method](https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#dom-authenticatorattestationresponse-transports-slot): :slightly_smiling_face:

>These values are the transports that the authenticator is believed to support, or an empty sequence if the information is unavailable.

If the `getTransports` method doesn't exist, then the information is indeed not available, so I think this should be an appropriate fallback value.

See also the definition of [`navigator.credentials.get()`](https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#ref-for-dom-publickeycredentialrequestoptions-allowcredentials%E2%91%A5) - any credential descriptor with an empty `transports` member will simply not affect the _distinctTransports_ variable being computed there.